web-common/src/features/custom-dashboards/Chart.svelte

```diff
@@ -0,0 +1,61 @@
+<script lang="ts">
+  import DashVegaRenderer from "@rilldata/web-common/features/custom-dashboards/DashVegaRenderer.svelte";
+  import { useChart } from "@rilldata/web-common/features/charts/selectors";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import { onDestroy, onMount } from "svelte";
+  import type { VisualizationSpec } from "svelte-vega";
+
+  const observer = new ResizeObserver((entries) => {
+    for (const entry of entries) {
+      const { width, height } = entry.contentRect;
+      clientHeight = height;
+      clientWidth = width;
+    }
+  });
+
+  export let chartName: string;
+
+  let clientHeight: number;
+  let clientWidth: number;
+  let container: HTMLDivElement;
+  let error: unknown = "";
+  let parsedVegaSpec: VisualizationSpec | undefined = undefined;
+
+  $: chart = useChart($runtime.instanceId, chartName);
+
+  $: vegaSpec = $chart?.data?.chart?.spec?.vegaLiteSpec;
+
+  $: try {
+    parsedVegaSpec = vegaSpec
+      ? (JSON.parse(vegaSpec) as VisualizationSpec)
+      : undefined;
+    error = "";
+  } catch (e: unknown) {
+    error = e;
+  }
+
+  onMount(() => {
+    observer.observe(container);
+  });
+
+  onDestroy(() => {
+    observer.disconnect();
+  });
+</script>
+
+<div
+  class="h-full w-full overflow-hidden pointer-events-none"
+  bind:this={container}
+>
+  {#if error}
+    <p>{error}</p>
+  {:else if !parsedVegaSpec}
+    <p>Chart not available</p>
+  {:else}
+    <DashVegaRenderer
+      spec={parsedVegaSpec}
```

Comment:
DashVegaRenderer seem to want `data` is there a layer missing here? Because of this, I dont see the chart at all even if it is valid.